test/aria/widgets/wai/tabs/update1/TabsJawsTest.js

```diff
@@ -0,0 +1,314 @@
+                add('_test');
+                add('_checkHistory');
+            }, this.end);
+        },
+
+
+
+        ////////////////////////////////////////////////////////////////////////
+        //
+        ////////////////////////////////////////////////////////////////////////
+
+        _test : function (callback) {
+            var data = this._getData();
+
+            var tabs = data.tabs;
+            var tabPanel = data.tabPanel;
+            var elementBefore = data.elementBefore;
+            var elementInside = data.elementInside;
+
+            this._executeStepsAndWriteHistory(callback, function (api) {
+                // ----------------------------------------------- destructuring
+
+                var step = api.addStep;
+                var entry = api.addToHistory;
+
+                // --------------------------------------------- local functions
+
+                var self = this;
+
+                function selectStartPoint() {
+                    step(['click', self.getElementById(elementBefore.id)]);
+                }
+
+                function getTabDescription(tab, key) {
+                    // ---------------------------------------------------------
+
+                    var isJawsNavigation = key === 'down';
+                    var description = [];
+
+                    // ---------------------------------------------------------
+
+                    if (isJawsNavigation) {
+                        if (tab.wrapper != null) {
+                            description.push(tab.wrapperText);
+                        } else {
+                            entry(tab.textContent);
+                        }
+                    } else {
+                        description.push(tab.title);
+                    }
+
+                    // ---------------------------------------------------------
+
+                    description.push('Tab');
+                    if (isJawsNavigation) {
+                        description.push('closed');
+                    }
+                    description.push('collapsed');
+                    if (!isJawsNavigation) {
+                        description.push('Use JawsKey+Alt+R to read descriptive text');
+                    }
+                    if (isJawsNavigation && tab.wrapper != null) {
+                        description.push(tab.textContent);
+                    }
+                    entry(description.join(' '));
+
+                    // ---------------------------------------------------------
+
+                    if (!isJawsNavigation) {
+                        if (tab.labelId) {
+                            entry(tab.textContent);
+                        }
+
+                        if (tab.description != null) {
+                            entry(tab.description);
+                        }
+                        entry('To activate tab page press Spacebar.');
+                    }
+                }
```

Comment:
For future tests, it is better (more readable and safer) to explicitly write the expected output rather than re-generating it from different parameters.